extensions/ql-vscode/src/debugger/debugger-factory.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  debug,
+  DebugAdapterDescriptor,
+  DebugAdapterDescriptorFactory,
+  DebugAdapterExecutable,
+  DebugAdapterInlineImplementation,
+  DebugConfigurationProviderTriggerKind,
+  DebugSession,
+  ProviderResult,
+} from "vscode";
+import { isCanary } from "../config";
+import { LocalQueries } from "../local-queries";
+import { DisposableObject } from "../pure/disposable-object";
+import { QueryRunner } from "../queryRunner";
+import { QLDebugConfigurationProvider } from "./debug-configuration";
+import { QLDebugSession } from "./debug-session";
+
+export class QLDebugAdapterDescriptorFactory
+  extends DisposableObject
+  implements DebugAdapterDescriptorFactory
+{
+  constructor(
+    private readonly queryStorageDir: string,
+    private readonly queryRunner: QueryRunner,
+    localQueries: LocalQueries,
+  ) {
+    super();
+
+    this.push(debug.registerDebugAdapterDescriptorFactory("codeql", this));
+    this.push(
+      debug.registerDebugConfigurationProvider(
+        "codeql",
+        new QLDebugConfigurationProvider(localQueries),
+        DebugConfigurationProviderTriggerKind.Dynamic,
+      ),
+    );
+
+    this.push(debug.onDidStartDebugSession(this.handleOnDidStartDebugSession));
```

Comment:
This method is not bound to `this`, so we wouldn't have access to `this` variables here:

```suggestion
    this.push(debug.onDidStartDebugSession(this.handleOnDidStartDebugSession.bind(this)));
```

Reply:
Removed because the handler didn't do anything anyway.